python/amici/ode_export.py

```diff
@@ -1854,8 +1870,39 @@ def _compute_equation(self, name: str) -> None:
                 sp.zeros(self.num_cons_law(), self.num_states_solver())
 
         elif name == 'dtcldp':
-            # force symbols
-            self._eqs[name] = self.sym(name)
+            self._derivative('total_cl', 'p', name=name)
+
+        elif name == 'stotal_cl':
+            # stotal_cl = dtotal_cl/dp +  dtotal_cl/dx_rdata * sx_rdata
+            # shape: ncl x np
+            dtotal_cldp = smart_jacobian(self.eq('total_cl'), self.sym('p'))
+            dtotal_cldx_rdata = smart_jacobian(self.eq('total_cl'),
+                                               self.sym('x_rdata'))
+            self._eqs[name] = dtotal_cldp
+            tmp = smart_multiply(dtotal_cldx_rdata, self.sym('sx_rdata'))
+            for ip in range(self._eqs[name].shape[1]):
+                self._eqs[name][:, ip] += tmp
+        elif name == 'sx_rdata':
+            # sx_rdata =  dx_rdata/dx_solver * sx_solver
+            #              + dx_rdata/d_tcl * stcl + dxrdata/dp
+            # shape: nx_rdata, 1
+            dx_rdata_dx_solver = smart_jacobian(self.eq('x_rdata'),
```

Comment:
if we don't have any cls, we can simply `self._eqs[name] = self.sym('sx_solver')`, will probably speed things up quite a bit for larger models.

Reply:
Thanks, makes sense